### PR TITLE
[5.0] Show proper error message when package is installed with no extensions

### PR DESCRIPTION
**Pull request by BrainforgeUK**
Avoid fatal PHP error on malformed manifest.

Pull Request for Issue # .

### Summary of Changes

Avoid exception when presented with a malformed manifest.

### Testing Instructions

Install a package without any extensions.

### Actual result BEFORE applying this Pull Request

Get fatal PHP exception.
count(): Argument #1 ($value) must be of type Countable|array, null given

### Expected result AFTER applying this Pull Request

Get message:
JLIB_INSTALLER_ABORT_PACK_INSTALL_NO_FILES

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by laoneo** [comment]
Do you have actually a package to test this?

**Comment by BrainforgeUK** [comment]
[pkg_test.zip](https://github.com/joomla/joomla-cms/files/13346135/pkg_test.zip)

Without the change this throws a system error.
Look at errors in browser console for more detail.

With the change get an unhelpful JSON error.
This is another Joomla issue I am not attempting to resolve here - less confusing if solution part of another pull request.
Refresh the screen and get the proper error message should have received instead of the JSON error.

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on c9428c93222e64d8d563100a33d39bcb0e199a89<br /><br />Installed the empty package and got a proper error message and a normal one (DPCalendar), which still worked.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42337">issues.joomla.org/tracker/joomla-cms/42337</a>.</sub>

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 9638a7ec03902bef4e588b568100be4c390a87d7<br /><br />Tested again a valid and empty package, now no JS errors anymore.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42337">issues.joomla.org/tracker/joomla-cms/42337</a>.</sub>

File: libraries/src/Installer/Adapter/PackageAdapter.php [hunk 1]
```diff
@@ -115,15 +115,30 @@ protected function checkExtensionInFilesystem()
...
+
+        if ($attributes) {
+            $folder = (string) $attributes->folder;
+
+            if ($folder) {
+                $source .= '/' . $folder;
+            }
+        }
+
+        // Install all necessary files
```
Comment (brianteeman) [inline libraries/src/Installer/Adapter/PackageAdapter.php:130]: is this needed? it looks like a copy of 141

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 9638a7ec03902bef4e588b568100be4c390a87d7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42337">issues.joomla.org/tracker/joomla-cms/42337</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42337">issues.joomla.org/tracker/joomla-cms/42337</a>.</sub>

File: libraries/src/Installer/Adapter/PackageAdapter.php (hunk 1 above)
Comment (Quy) [reply libraries/src/Installer/Adapter/PackageAdapter.php:130]: @BrainforgeUK Please update so this can be set to RTC. Thanks.

Comment (BrainforgeUK) [reply libraries/src/Installer/Adapter/PackageAdapter.php:130]: Removed accidently duplicated code.

**Comment by bembelimen** [comment]
Thx

**Pull request merged**